### PR TITLE
Fix `IntroScreen` retrieving and iterating all realm beatmap sets

### PR DESCRIPTION
**Pull request by peppy**
Really want to remove the method completely, but it's used in a lot of tests and can't easily be made into a test-only extension method (unless `ContextFactory` is exposed in `BeatmapManager` as `public`).

This was the last usage of the method outside of tests.

- [x] Depends on https://github.com/ppy/osu/pull/16535.

### CODE REVIEW

File: osu.Game/Screens/Menu/IntroScreen.cs [hunk 1]
```diff
@@ -93,57 +94,56 @@ private void load(OsuConfigManager config, Framework.Game game, RealmContextFact
...
-                if (sets.Count > 0)
+                realmContextFactory.Run(realm =>
                 {
-                    setInfo = beatmaps.QueryBeatmapSet(s => s.ID == sets[RNG.Next(0, sets.Count - 1)].ID);
-                    setInfo?.PerformRead(s =>
+                    var sets = realm.All<BeatmapSetInfo>().Where(s => !s.DeletePending && !s.Protected).AsRealmCollection();
+
+                    int setCount = sets.Count;
+
+                    if (sets.Any())
```
Comment (frenzibyte) [inline osu.Game/Screens/Menu/IntroScreen.cs:106]: nit:
```suggestion
                    if (setCount > 0)
```

**Comment by bdach** [comment]
Looks like tests need attention here, `TestSceneScreenNavigation` and `TestSceneMusicActionHandling` have universal failures.

**Comment by peppy** [comment]
Weird, that must be fixed in my follow-up PR in this chain. Might be easier to commit the extra commits here, will check.

**Comment by peppy** [comment]
Turns out I just got the nesting completely wrong in a rebase or something. Should work as expected now.

**Review by frenzibyte — APPROVED** [review]
Good stuff, tested to work with all different intro settings combinations and scenarios. (~~the only difference I can see is that master might use a protected/intro beatmap in the `!MainMenu.Value` case; but excluding protected makes sense anyways, since you don't want other intro themes to play in the current intro~~ in fact, it's actually a bug from after migrating `BeatmapManager` to realm that's now fixed here)

**Review by bdach — APPROVED** [review]

**Pull request merged**